public/js/p3/widget/viewer/SubSystemMap.js

```diff
@@ -15,40 +15,67 @@ define([
 		subsystemClass: "",
 		subclass: "",
 		
-		genomeIds: "",
-		subsystemId: "",
 		taxonId: "",
+		displayDefaultGenomes: false,
 
 		onSetState: function(attr, oldVal, state){
 
 			if(!state){
 				return;
 			}
 
-			var subsystemData = this.getStateParams(state);
+			var display_reference_genomes = this.getStateParams(state);
 
-			state.genome_ids = subsystemData.genome_ids;
-			state.genome_ids_arr = subsystemData.genome_ids.split(',');
-			state.subsystem_id = subsystemData.subsystem_id;
-			state.selectionData = subsystemData.selectionData;
+			var that = this;
+			var query = "?ne(genome_id," + state.genome_ids_without_reference + ")&eq(taxon_lineage_ids,2)&eq('reference_genome','Reference'),&select(genome_id,genome_name,reference_genome)&limit(25000)";
+			return when(request.get(PathJoin(window.App.dataAPI, "genome", query), {
+				headers: {
+					'Accept': "application/json",
+					'Content-Type': "application/rqlquery+x-www-form-urlencoded"
+				},
+				handleAs: "json"
+			}), function(response){
 
-			var self = this;
-			when(this.getGenomeIdsBySubsystemId(state.genome_ids_arr, state.subsystem_id), function(genomeIds){
-				state.genome_ids = genomeIds;
-				self.viewer.set('visible', true);
-			});
+				var reference_genome_ids = response.map(function(genome){
+					return genome.genome_id;
+				})
+
+				state.reference_genome_ids_only = reference_genome_ids;
+
+				that.state.genome_ids.forEach(function(genome_id){
+					reference_genome_ids.unshift(genome_id);
+				});
 
-			// when(this.getSubsystemDescription(subsystemData.subsystem_id), function(description){
-			// 	//state.description = description	
-			// 	$('#subSystemHeatmap').append( "<p>" + description + "</p>" );
-			// 	$('#subSystemHeatmap').css("height", "70px");
-			// });
+				state.genome_ids_with_reference = reference_genome_ids;
+				state.genome_ids = reference_genome_ids;
 				
-			window.document.title = 'Subsystem Map';
+				when(that.getGenomeIdsBySubsystemId(that.state.genome_ids, that.state.subsystem_id), function(genomeIds){
+					that.viewer.set('visible', true);
+					$('#subsystemheatmap').attr('style','height: 170px');
```

Comment:
Yes - the header content is determined by data returned in this call and it is important to not show the module until that data is available.

Reply:
no i meant, adding height in here. I think since we specified the height in content pane, we may not need setting extra style here. But I haven't tested though. i just wanted to keep the changes minimal in order to keep code clean.